Formula/e/edgevpn.rb

```diff
@@ -0,0 +1,27 @@
+class Edgevpn < Formula
+  desc "Immutable, decentralized, statically built p2p VPN"
+  homepage "https://mudler.github.io/edgevpn"
+  url "https://github.com/mudler/edgevpn/archive/refs/tags/v0.24.6.tar.gz"
+  sha256 "cc7fc9e15273e374cbf4ae72aa2d95748a0af36d92f3d16a0f26fc8c07ffd497"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.date=#{time.iso8601}
+    ].join(" ")
+
+    ENV["CGO_ENABLED"] = "0"
+    system "go", "build", *std_go_args(ldflags: ldflags)
+
+    # generate_completions_from_executable(bin/"edgevpn", "completion")
+  end
+
+  test do
+    help_output = pipe_output("#{bin}/edgevpn help")
+    assert_match "edgevpn - edgevpn --config /etc/edgevpn/config.yaml", help_output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the feedback and sorry that I missed that in the docs. I made a test to validate that edgevpn can generate a token